### PR TITLE
Update Dockerfile and docker-compose

### LINKED ISSUE #37169: Update Dockerfile and docker-compose in .devcontainer directory
The current implementation of the Docker Image in/.devcontainer uses PHP7.4, which is no longer compatible with the minimal PHP requirement of PHP 8.0 for developers trying to set up a local development environment with docker. Updating (and maybe tweaking a bit) the current implementation would help developers interested in contributing to set up their local environment quicker. What do you think?

### PR DESCRIPTION
**Pull request by sorbaugh**
* Resolves: #37169

## Summary
Proposed change to in-built docker-compose.yml and Dockerfile. The current implementation uses php 7.4, which is no longer compatible with the required PHP version of the server. I upped this to PHP 8.1

List of changes:
- Upped PHP Version to 8.1
- QoL: Added Apache Webserver so the Container works "out of the box" after docker-compose up -d
- Mounting whole project as volume to /var/www/html in docker-compose.yml (and set WORKDIR to /var/www/html)

Tested in a Docker for Windows environment.

How to run: 
- Navigate to .devcontainer directory in terminal and execute 
`docker-compose up -d`
- After container is started. Restart Apache Service in Container
`docker exec devcontainer-nextclouddev-1 service apache2 restart`
- Initialize Submodules (Only necessary for first time setup)
`docker exec devcontainer-nextclouddev-1 git submodule update --init`

After Container starts, open in Browser:
`http://localhost/`

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by R0Wi** [comment]
LGTM so far. I added some minor improvements in 6bfa5d42485820cd5712ecf23f235f4d61f5031c . It should now be possible to just open a fresh Codespaces instance and start debugging right away. Maybe you want to pick this commit to your branch as well?

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -4,31 +4,43 @@ ARG DEBIAN_FRONTEND=noninteractive
 
 # PHP
 RUN apt-get update -y
+
+RUN apt install -y apache2 vim software-properties-common
+
+RUN add-apt-repository ppa:ondrej/php
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply .devcontainer/Dockerfile:10]: this is using focal
image

[the remark this answers is not in the document]
Comment (szaimen) [reply .devcontainer/Dockerfile:10]: Ah I meant jammy

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -44,4 +56,6 @@ RUN add-apt-repository \
    stable"
 RUN apt-get update -y
 RUN apt-get install -y docker-ce docker-ce-cli containerd.io
-RUN ln -s /var/run/docker-host.sock /var/run/docker.sock  
+RUN ln -s /var/run/docker-host.sock /var/run/docker.sock
```
Comment (artonge) [inline .devcontainer/Dockerfile:59]: Any reason this is needed ?
Not even sure why we are binding the docker socket in the first place.

File: .devcontainer/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (R0Wi) [reply .devcontainer/Dockerfile]: Resolved by e6996140bf28ccb8e08f31f27dc5a6ee860036d1

**Comment by R0Wi** [comment]
I added the following:

-  211c5ec398a89413a943b432051b89585ed349bc : I'm now able to directly open and debug the NC project after codespaces start without any futher steps
- e6996140bf28ccb8e08f31f27dc5a6ee860036d1 : Use the official jammy base image to be able to use PHP 8.1 without 3rd party repos

**Review by R0Wi — APPROVED** [review]

**Comment by R0Wi** [comment]
@sorbaugh any reason why this has been closed?

**Pull request closed without merging**